packages/zostso/src/ReceiveTsoApp.ts

```diff
@@ -0,0 +1,91 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { AbstractSession } from "@zowe/imperative";
+import { ZosmfRestClient } from "@zowe/core-for-zowe-sdk";
+import { TsoValidator } from "./TsoValidator";
+import { noAccountNumber, TsoConstants } from "./TsoConstants";
+import { ITsoAppCommunicationParms } from "./doc/input/ITsoAppCommunicationParms";
+import { IASAppResponse } from "./doc/IASAppResponse";
+import { DEFAULT_SPINNER_CHARS } from "@zowe/imperative";
+
+export class ReceiveTsoApp {
+    public static async receive(
+        session: AbstractSession,
+        accountNumber: string,
+        params: ITsoAppCommunicationParms
+    ): Promise<IASAppResponse> {
+        TsoValidator.validateSession(session);
+        TsoValidator.validateNotEmptyString(accountNumber, noAccountNumber.message);
+        const TIMEOUT_SECONDS: number = params.timeout;
+        const spinnerChars = DEFAULT_SPINNER_CHARS.split("");
+        let spinnerIndex = 0;
+
+        // Start the spinner
+        const spinner = setInterval(() => {
+            process.stdout.write(`\rReceiving response... ${spinnerChars[spinnerIndex]}`);
+            spinnerIndex = (spinnerIndex + 1) % spinnerChars.length;
+        }, 100);
```

Comment:
Very clever way to implement your own spinner,

I'm curious if we should this to a set of public methods in the mProgressApi in-line class defined in the `.response.progress` getter.
https://github.com/zowe/zowe-cli/blob/master/packages/imperative/src/cmd/src/response/CommandResponse.ts#L691